src/lib/block/serpent/serpent_fn.h

```diff
@@ -10,9 +10,19 @@
 #include <botan/compiler.h>
 #include <botan/types.h>
 #include <botan/internal/rotate.h>
+#include <concepts>
 
 namespace Botan::Serpent_F {
 
+// Concept for types that support bitwise operations (unsigned integers or SIMD types)
+template <typename T>
+concept BitsliceT = std::unsigned_integral<T> || requires(T& a, const T& b) {
+   a ^= b;
+   a &= b;
+   a |= b;
+   ~a;
+};
```

Comment:
Properly nitpicking here, but: `std::unsigned_integral` can do all the things you spelled out in the requires clause. You might as well leave it out as an explicit option. 🤷  Also: the `T`-suffix is somewhat against typical convention where concepts have more "natural" names. Maybe "bislicable" (what a mess), or "bitwise_operand", meh. Not sure.

Reply:
Right but its an or here since the simd wrappers are not integral types